baselines/packages/mimir/test/no-restricted-property-access/default/test.ts.lint

```diff
@@ -0,0 +1,299 @@
+
+declare let publicPrivate: Public & Private;
+publicPrivate['prop'];
+~~~~~~~~~~~~~~~~~~~~~  [error no-restricted-property-access: Property 'prop' has conflicting declarations and is inaccessible in type 'Public & Private'.]
+
+declare function decorator(...args: any[]): any;
+
+class A {
+    protected prop = 1;
+
+    method(a: A, b: B, c: C) {
+        a['prop'];
+        b['prop'];
+        c['prop'];
+    }
+}
+
+class B extends A {
+    method(a: A, b: B, c: C) {
+        a['prop'];
+        ~~~~~~~~~  [error no-restricted-property-access: Property 'prop' is protected and only accessible through an instance of class 'B'.]
+        b['prop'];
+        c['prop'];
+    }
+}
+
+class C extends B {
+    method(a: A, b: B, c: C) {
+        a['prop'];
+        ~~~~~~~~~  [error no-restricted-property-access: Property 'prop' is protected and only accessible through an instance of class 'C'.]
+        b['prop'];
+        ~~~~~~~~~  [error no-restricted-property-access: Property 'prop' is protected and only accessible through an instance of class 'C'.]
+        c['prop'];
+    }
+}
+
+function fnA(this: A, a: A, b: B, c: C) {
+    a['prop'];
+    b['prop'];
+    c['prop'];
+}
+function fnB(this: B, a: A, b: B, c: C) {
+    a['prop'];
+    ~~~~~~~~~  [error no-restricted-property-access: Property 'prop' is protected and only accessible through an instance of class 'B'.]
+    b['prop'];
+    c['prop'];
+}
+function fnC(this: B, a: A, b: B, c: C) {
+    a['prop'];
+    ~~~~~~~~~  [error no-restricted-property-access: Property 'prop' is protected and only accessible through an instance of class 'B'.]
+    b['prop'];
+    c['prop'];
+}
+
+type Constructor<T = {}> = new (...args: any[]) => T;
+
+function Mixin<TBase extends Constructor>(Base: TBase) {
+    abstract class C extends Base {
+        abstract method(): number;
+    }
+    return C;
+}
+
+class MixinSubclass extends Mixin(class {a() {return 1;}}) {
+    method() {
+        return super['method']();
+               ~~~~~~~~~~~~~~~    [error no-restricted-property-access: Abstract method 'method' in class 'C' cannot be accessed via the 'super' keyword.]
+    }
+}
+
+class MixinSubclass2 extends Mixin(class {method() {return 1;}}) {
+    prop = this['method']();
+    other() {
+        return super['method']();
+               ~~~~~~~~~~~~~~~    [error no-restricted-property-access: Abstract method 'method' in class 'C' cannot be accessed via the 'super' keyword.]
+    }
+}
+
+declare var tuple: [string, string];
+tuple['length']; // don't crash
+
+class WithMethod {
+    method() { return 1; }
+}
+abstract class WithAbstractMethod {
+    abstract method(): number;
+}
+declare const Base: new() => WithMethod & WithAbstractMethod;
+class IntersectionSubclass extends Base {
+    doStuff() {
+        return super['method']();
+               ~~~~~~~~~~~~~~~    [error no-restricted-property-access: Abstract method 'method' in class 'WithAbstractMethod' cannot be accessed via the 'super' keyword.]
```

Comment:
Does this really make sense? One of the intersecting classes has an implementation.